core/src/main/scala-2/spire/random/Op.scala

```diff
@@ -0,0 +1,47 @@
+package spire
+package random
+
+import scala.collection.Factory
+
+sealed trait Op[+A] {
+
+  def flatMap[B](f: A => Op[B]): Op[B] =
+    this match {
+      case FlatMap(a, g) => FlatMap(a, (x: Any) => g(x).flatMap(f))
```

Comment:
```suggestion
      case FlatMap(a, g) => FlatMap(a, (x: Any) => g(x).flatMap(f))
      case fm: FlatMap[a, _] => FlatMap(fm.sub, (x: a) => fm.k(x).flatMap(f))
```

This line should make cross compilation work between `2.13` and `3.0`.